core/rfb.js

```diff
@@ -542,7 +544,17 @@
                     Util.Error("Got data while disconnected");
                     break;
                 case 'normal':
-                    while (this._normal_msg() && this._sock.rQlen() > 0);
+                    while (true) {
+                        if (this._flushing) {
+                            break;
+                        }
+                        if (!this._normal_msg()) {
```

Comment:
so, here we're replacing a long render queue for a bigger receive queue.  That's fine *until* we hit the point at which we need to resize the receive queue, at which point we take a performance hit (especially if we have to resize the receive queue multiple times -- allocating large typed arrays is fairly costly).  It's mitigated by the fact that we never downsize our arrays, but I'm still concerned, and I'd like to see some numbers on this change from different VNC sessions, with different servers (at least qemu's VNC server and TigerVNC).